src/Config.cc

```diff
@@ -570,6 +571,18 @@ std::vector<ZyppFlags::CommandGroup> Config::cliOptions()
               // translators: --gpg-auto-import-keys
               _("Automatically trust and import new repository signing keys.")
         },
+        { "gpg-auto-import-key-id", 0, ZyppFlags::RequiredArgument, std::move( ZyppFlags::StringType( &gpg_auto_import_key_id, "", "KEY_ID").
```

Comment:
It's IMO debatable whether a new option is introduced or `gpg-auto-import-keys` is amended to take an optional argument. In any case a key or `,`-separated list of keys should be accepted.

Reply:
Great, I will make sure to accept comma separated keys as well.

I also like your suggestion of amending `gpg-auto-import-keys` rather than adding a new parameter. Do you want to open up that conversation to other stakeholders, or can we move forward with the single parameter implementation?

Reply:
We can move on. Staying with `gpg-auto-import-key` and amending it to 

1. take an optional argument (comma separated list of IDs)
2. allow multiple occurrences of  `gpg-auto-import-key`

should be feasible for the argparser .

This enables the user to use `--gpg-auto-import-key=A,B` or `--gpg-auto-import-key=A --gpg-auto-import-key=B` as they like it.

It also allows to solve the issue of `--gpg-auto-import-key  ... --gpg-auto-import-key=A` in a natural way. No errors about conflicting options are needed. The options build up a set of key IDs. An empty set accepts all keys as it currently does. Otherwise only the keys in the set are auto-accepted. Keys not in the set are treated as usual.

(@bzeller JFYI)

Reply:
In order to make `gpg-auto-import-keys` optional, while still accepting comma separated lists of IDs as you suggest in your comment here, I had to make a small adjustment to the GenericContainerType definition in the argparser. If there was a better way to go about achieving this, let me know and I would be happy to make adjustments if necessary.

The change to GenericContainerType is in [its own commit](https://github.com/openSUSE/zypper/pull/421/commits/e21b6e9ac136b734a78d16413cc9a31ed204fc98).